runtime_api.go

```diff
@@ -0,0 +1,183 @@
+		// hidden, otherwise forwards arguments to printany to print them
+		// as normal. Although we add an if statement to printany to do
+		// nothing if panics are hidden, printany only takes one argument,
+		// and both print and printany are used to print panic messages.
+		// panicprint's entire purpose is to act as a replacement to print
+		// that respects HidePanics, and print is variadic, so print must be
+		// replaced by a variadic function, hence panicprint.
+		//
+		// We will also add two statements to printany:
+		// 1. An if statement that returns early if panics are hidden
+		// 2. An additional case statement that handles printing runtime.hex
+		//    values. Without this case statement, the default case will print
+		//    the runtime.hex values in a way not consistent with normal panic
+		//    outputs
+		for _, decl := range file.Decls {
+			funcDecl, ok := decl.(*ast.FuncDecl)
+			if ok && funcDecl.Name.Name == "printany" {
+				for _, stmt := range funcDecl.Body.List {
+					switchStmt, ok := stmt.(*ast.TypeSwitchStmt)
+					if !ok {
+						continue
+					}
+
+					switchStmt.Body.List = append(switchStmt.Body.List, printanyHexCase)
+					break
+				}
+
+				funcDecl.Body.List = append([]ast.Stmt{hidePanicsCheckStmt}, funcDecl.Body.List...)
+				break
+			}
+		}
+
+		file.Decls = append(file.Decls, panicprintDecl)
+	case "panic.go", "traceback.go":
+		// Change all calls to print, which we don't control, to
+		// panicprint, which we do control and does the same thing.
+		ast.Inspect(file, switchPanicPrints)
+	}
+}
+
+var hidePanicsCheckStmt = &ast.IfStmt{
+	Cond: &ast.Ident{Name: "hidePanics"},
+	Body: &ast.BlockStmt{List: []ast.Stmt{
+		&ast.ReturnStmt{},
+	}},
+}
+
+var hidePanicsDecls = []ast.Decl{
+	&ast.GenDecl{
+		Tok: token.VAR,
+		Specs: []ast.Spec{
+			&ast.ValueSpec{
+				Names: []*ast.Ident{
+					&ast.Ident{Name: "hidePanics"},
+				},
+				Type: &ast.Ident{Name: "bool"},
+			},
+		},
+	},
+	&ast.FuncDecl{
+		Name: &ast.Ident{Name: "HidePanics"},
+		Type: &ast.FuncType{
+			Params: &ast.FieldList{
+				List: []*ast.Field{
+					&ast.Field{
+						Names: []*ast.Ident{
+							&ast.Ident{Name: "hide"},
+						},
+						Type: &ast.Ident{Name: "bool"},
+					},
+				},
+			},
+		},
+		Body: &ast.BlockStmt{
+			List: []ast.Stmt{
+				&ast.AssignStmt{
+					Lhs: []ast.Expr{
+						&ast.Ident{Name: "hidePanics"},
+					},
+					Tok: token.ASSIGN,
+					Rhs: []ast.Expr{
+						&ast.Ident{Name: "hide"},
+					},
+				},
+			},
+		},
+	},
+}
+
+var printanyHexCase = &ast.CaseClause{
+	List: []ast.Expr{
+		&ast.Ident{Name: "hex"},
+	},
+	Body: []ast.Stmt{
+		&ast.ExprStmt{
+			X: &ast.CallExpr{
+				Fun: &ast.Ident{Name: "print"},
+				Args: []ast.Expr{
+					&ast.Ident{Name: "v"},
+				},
+			},
+		},
+	},
+}
+
+var panicprintDecl = &ast.FuncDecl{
+	Name: &ast.Ident{Name: "panicprint"},
+	Type: &ast.FuncType{Params: &ast.FieldList{
+		List: []*ast.Field{
+			&ast.Field{
```

Comment:
join with the above; also run `gofmt -l -w -s` to remove redundant types like this one